### PR TITLE
Add bertopic

### PR DESCRIPTION
**Pull request by mimshiran**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mimshiran** [comment]
@MaartenGr please confirm that you are a maintainer. Thanks

**Comment by MaartenGr** [comment]
@mimshiran Installing all optional dependencies at once is likely to cause some issues as they can diverge quite a bit. It might be best to either skip over all optional dependencies and allow users to install those themselves, or follow along https://github.com/conda/conda/issues/7502 where you create multiple packages but I am not a big fan of the latter. Perhaps starting from a barebones solution without, for example, gensim might help. But then again, not that familiar with building conda recipes 😅

**Comment by MaartenGr** [comment]
> @MaartenGr please confirm that you are a maintainer. Thanks

Yep, glad to help out!

**Comment by mimshiran** [comment]
PR is ready. Request for review/merging.

cc: @conda-forge/help-python

Reason for Windows build failure:

The package `sentence-transformers` requires `pytorch`, which is only available for `linux` and `osx` on conda-forge, so the windows build fails.

**Comment by mimshiran** [comment]
> @mimshiran Installing all optional dependencies at once is likely to cause some issues as they can diverge quite a bit. It might be best to either skip over all optional dependencies and allow users to install those themselves, or follow along [conda/conda#7502](https://github.com/conda/conda/issues/7502) where you create multiple packages but I am not a big fan of the latter. Perhaps starting from a barebones solution without, for example, gensim might help. But then again, not that familiar with building conda recipes 😅

I had some problems with sentence-transformers for windows. linux and osx were successful. so I'm waiting for them to check it.

**Comment by mimshiran** [comment]
PR is ready. Request for review/merging.

cc: @conda-forge/help-python
cc: @conda-forge/staged-recipes

Reason for Windows build failure:

The package sentence-transformers requires `pytorch`, which is only available for `linux` and `osx` on conda-forge, so the windows build fails.

File: recipes/bertopic/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python >=3.7
+  run:
+    - python >=3.7
+    - pyyaml <6.0
```
Comment (jaimergp) [inline recipes/bertopic/meta.yaml:24]: Let's consider this temporary until upstream has fixed their issues. Could you please add a comment here so we don't forget? Thanks!

Comment (mimshiran) [reply recipes/bertopic/meta.yaml:24]: Do you mean we should exclude Windows for the time being? Thanks!

**Comment by sugatoray** [comment]
Hi @mimshiran and @MaartenGr, 

I started adding the package `stripnet` (PR: https://github.com/conda-forge/staged-recipes/pull/17460) a few days after you started working on `bertopic`. But since, `bertopic` is a dependency of `stripnet`, I had to add the recipe file for bertopic in the same PR as well. At the time I was unaware of your endeavor on this PR. 

@jaimergp is also one of the reviewers on my PR. Until bertopic is merged, stripnet cannot be build without also shipping bertopic alongside in the same PR (mine). Both recipes are ready.

So, would like to be on the maintainer list of my PR for bertopic (we will be co-maintainers of bertopic-feedstock)? In that case, a single merge will push both packages to conda-forge. The PR as I believe is ready to merge (unless the reviewers have any additional concerns/suggestions).

Thank you.

File: recipes/bertopic/meta.yaml (hunk 1 above)
Comment (jaimergp) [reply recipes/bertopic/meta.yaml:24]: No, I meant the `pyyaml` version constrain (<6) should be considered a temporary workaround and if possible would be removed in the future when upstream has fixed their compatibility issues. Adding a comment saying `# This is a temporary workaround; review in the future` should be enough:
```suggestion
    # This is a temporary workaround; review in the future
    - pyyaml <6.0
```

**Comment by mimshiran** [comment]
@sugatoray @MaartenGr 

This PR is ready to be merged. Once it's merged you can add it as a dependency to your project. So there's no need to change things here. You can simply remove the bertopic folder from your PR and things should work.

Happy new year!

**Comment by mimshiran** [comment]
@ jaimergp I have committed your suggested changes, please review and let me know if you have any other suggestions.

**Comment by sugatoray** [comment]
@mimshiran In that case, please add me as a maintainer here. Thank you.

File: recipes/bertopic/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+about:
+  home: https://github.com/MaartenGr/BERTopic
+  summary: BERTopic performs topic Modeling with state-of-the-art transformer models.
+  license: MIT
+  license_file: LICENSE
+
+extra:
+  recipe-maintainers:
+    - mimshiran
+    - MaartenGr
```
Comment (sugatoray) [inline recipes/bertopic/meta.yaml:53]:
```suggestion
    - MaartenGr
    - sugatoray
```

**Pull request merged**